### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by azrsh**
- A minimal action implementation that sets up sorah-ruby.

## Features

- Install the package according to the platform from [Software Repositories - sorah.jp](https://sorah.jp/packaging/debian/).
- Cache RubyGems installed with Bundler.

## Specification

- Takes the following variables as input.
   - `ruby-version`
   - `ruby-version-file`
   - `bundler-cache`
- `ruby-version` and `ruby-version-file` are exclusive inputs and only one of them can be specified
- If neither `ruby-version` nor `ruby-version-file` are specified, use `ruby-version-file` = `.ruby-version` as the default value
- The value of `ruby-version` or the content of `ruby-version-file` can be any of the following versions. It must be a string.
   - `3.0`
   - `3.1`
   - `3.2`
- `bundler-cache` must be a boolean. Default value is false.
- When `bundler-cache` is true, automatically run `bundle install` after restoring RubyGems installed by Bundler from GitHub Actions cache. If lockfile has changed, update the cache.
- A lockfile must exist in the target directory.
- Environment variables are passed through to Bundler, so any environment variable can change the behavior of Bundler.

## Notes

- Ruby versions below 2.7 are already at their end of life, so I didn't support them. Add support if needed.
- This action only accepts `3.0` `3.1` `3.2` as Ruby version, but rbenv does not accept such version specification by default. So we might want to accept the notation to specify the patch version to use generic `.ruby-version` files for this action.
- Unable to test this action on CI for environments not in the GitHub hosted runner. Any ideas?

### CODE REVIEW

File: src/platform.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as os from "node:os";
+import * as linuxOSRelase from "./util/linux_os_release";
```
Comment (sorah) [inline src/platform.ts]: typo

**Review by sorah — CHANGES_REQUESTED** [review]
- Write a README to explain the usage, otherwise I cannot review appropriately.
- Declare a MIT license please
- `rubyPackageVersionPrefix` shouldn't be in this actions code. It's painful to maintain `rubyPackageVersionPrefix`. For Actions purpose I think this can be flexible (always use the latest available on the apt repository), and having a option to use spefiic version would be better. Also, if we put these data in the action, users have to upgrade actions itself to install the updated packages.
- It'd be better if we can have less static data in this actions code. For instance, it's also painful to maintain distro availability data. Attempt to install unsupported setup should be notified through apt command error to users, so I don't think this action doesn't necessary to validate in advance. If you wish to do, parse repository or use `apt-cache` command to verify package existence. I think validation is overkill though.
- ruby-defaults package should be pinned, as done in Dockerfiles: https://github.com/sorah-rbpkg/dockerfiles/commit/5f976e61d13967eb4203299253d6ca17d466f080, to prevent unintended upgrade.

**Comment by sorah** [comment]
hint, apt-get install supports glob: `apt-get install 'ruby3.1=3.1.0-*'`

**Comment by azrsh** [comment]
> * Write a README to explain the usage, otherwise I cannot review appropriately.

Write README.md ab43dc1 d99ff5d 605e07a

> * Declare a MIT license please

Set LICENSE file 3e2343f

> * `rubyPackageVersionPrefix` shouldn't be in this actions code. It's painful to maintain `rubyPackageVersionPrefix`. For Actions purpose I think this can be flexible (always use the latest available on the apt repository), and having a option to use spefiic version would be better. Also, if we put these data in the action, users have to upgrade actions itself to install the updated packages.

Remove `rubyPackageVersionPrefix` 81c9ec9 and Add `ruby-package-version` option c579892

> * It'd be better if we can have less static data in this actions code. For instance, it's also painful to maintain distro availability data. Attempt to install unsupported setup should be notified through apt command error to users, so I don't think this action doesn't necessary to validate in advance. If you wish to do, parse repository or use `apt-cache` command to verify package existence. I think validation is overkill though.

Remove platform validation 93858c4

> * ruby-defaults package should be pinned, as done in Dockerfiles: [sorah-rbpkg/dockerfiles@5f976e6](https://github.com/sorah-rbpkg/dockerfiles/commit/5f976e61d13967eb4203299253d6ca17d466f080), to prevent unintended upgrade.

Pin ruby-defaults package 05b1342

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    runs-on: ["${{ matrix.runner }}"]
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: ./
+        with:
+          ruby-version: ${{ matrix.version }}
+
+      - run: |
+          ruby --version | grep '${{ matrix.version }}'
```
Comment (sorah) [inline .github/workflows/test.yml]: It's also worth to verify rubyN.M and ruby-defaults package through `dpkg -l`

Comment (sorah) [inline .github/workflows/test.yml]: consider having `^` so avoid ruby 3.2.7 to match 2.7 from matrix.version?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# sorah-rbpkg/actions
+
+## Features
+
+- Install the package according to the platform from [Software Repositories - sorah.jp](https://sorah.jp/packaging/debian/).
```
Comment (sorah) [inline README.md]:
```suggestion
- Install Ruby packages from https://sorah.jp/packaging/debian/.
```

File: apt_install.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash -eux
+
+sudo apt-get purge ruby || true
```
Comment (sorah) [inline apt_install.sh:4]: why?

File: apt_install.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+sudo apt-get update
+sudo apt-get install -y ca-certificates
+
+sudo mkdir -p /usr/local/share/keyrings/
+sudo cp $(dirname ${0})/sorah-ruby.gpg /usr/local/share/keyrings/
```
Comment (sorah) [inline apt_install.sh]: `install -D -m0644 -oroot -groot $(dirname ${0})/sorah-ruby.gpg /usr/local/share/keyrings/sorah-ruby.gpg` and remove `mkdir -p`

File: README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+You can explicitly specify the Ruby package version using the input `ruby-package-version`.
+
+```yaml
+steps:
+  - uses: actions/checkout@v3
+  - uses: sorah-rbpkg/actions@v1
+    with:
+      ruby-version: '3.1'
+      ruby-package-version: 3.1.3-0nkmi1~jammy
```
Comment (sorah) [inline README.md]: We can determine `ruby-version` from source version part of `ruby-package-version` and no one wants to specify debian version. Could you change the code to accept `X.Y.Z` format in `ruby-version`?

**Comment by sorah** [comment]
Please add support of 2.7 which just transitioned to EOL…

File: .github/workflows/test.yml (hunk 2 above)
Comment (azrsh) [reply .github/workflows/test.yml]: fixed 48080a9

Comment (azrsh) [reply .github/workflows/test.yml]: fixed 48080a9
Is this change in line with your intentions?

File: apt_install.sh (hunk 5 above)
Comment (azrsh) [reply apt_install.sh]: fixed 7e616b9

File: apt_install.sh (hunk 4 above)
Comment (azrsh) [reply apt_install.sh:4]: The GitHub hosted runners come with the `ruby` package installed by default, so trying to install the `ruby` package with this action will cause a conflict.

**Comment by azrsh** [comment]
Support Ruby 2.7 b3364f3 b568db9 8e73c34

File: README.md (hunk 6 above)
Comment (azrsh) [reply README.md]: Support patch version 43f6ceb cfe34d7 a124125

File: apt_install.sh (hunk 4 above)
Comment (sorah) [reply apt_install.sh:4]: Could you leave this as a comment?

**Review by sorah — APPROVED** [review]

File: apt_install.sh (hunk 4 above)
Comment (azrsh) [reply apt_install.sh:4]: commented eb19b58

**Comment by azrsh** [comment]
Cleaned up commits eb19b58 -> 6079d60 (no diff)

**Comment by azrsh** [comment]
Thank you for your review, @sorah !

**Pull request merged**